isthmus/src/main/java/io/substrait/isthmus/TypeConverter.java

```diff
@@ -56,6 +57,17 @@ public NamedStruct toNamedStruct(RelDataType type) {
     return NamedStruct.of(names, struct);
   }
 
+  public NamedStruct toNamedStruct(Map<String, RelDataType> nameToTypeMap) {
```

Comment:
I'm not sure if it's a good idea expose a public API like this. Because this takes a Map, the order of the fields in the final named struct is dependent on the iteration order of the Map.

Reply:
Changed